lib/libmbedtls/core/tee_lmd_provider.c

```diff
@@ -2049,13 +2116,26 @@ void crypto_cipher_free_ctx(void *ctx, uint32_t algo __maybe_unused)
 	 * could never have succeded above.
 	 */
 	assert(!cipher_get_ctx_size(algo, &ctx_size));
-	mbedtls_cipher_free(ctx);
+#if defined(CFG_CRYPTO_XTS) || defined(CFG_CRYPTO_CTS)
```

Comment:
Unless `CFG_CRYPTO_XTS` or `CFG_CRYPTO_CTS` is defined we'll potentially leak memory.
I see no harm in always having the `if (...)` below

Reply:
Yes, it is a mistake. Should not add them here.